dev/reference/transactions/transaction-pessimistic.md

```diff
@@ -0,0 +1,98 @@
+
+- Use the `BEGIN PESSIMISTIC;` statement to start the transaction in the pessimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.
+
+- Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the transactions processed in this session to be in the pessimistic transaction mode.
+
+- Enable the pessimistic transaction mode in the configuration file. This allows all transactions (except auto-committed single-statement ones) to adopt the pessimistic transaction mode.
+
+    ```
+    [pessimistic-txn]
+    enable = true
+    default = true
+    ```
+
+If the pessimistic transaction mode is enabled in the configuration file by default, use one of the following methods to adopt the optimistic transaction mode for the transaction:
+
+- Use the `BEGIN OPTIMISTIC;` statement to start the transaction in the optimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 OPTIMISTIC */;` to make it compatible with the MySQL syntax.
+
+- Execute the `set @@tidb_txn_mode = 'optimistic';` statement to allow all the transactions processed in this session to be in the optimistic transaction mode.
+
+## Enablement priority
+
+The three methods to enable the transaction mode are ordered from highest priority to lowest as follows:
+
+- Use `BEGIN PESSIMISTIC;` or `BEGIN OPTIMISTIC;`.
+
+- Set the session variable `tidb_txn_mode`.
+
+- Configure the parameter `default` in the configuration file. If you use a regular `BEGIN` statement and set the value of `tidb_txn_mode` to an empty string, then you can use `default` to determine whether to enable the pessimistic or optimistic transaction mode.
+
+## Configuration parameter
+
+The related configuration parameters are under the `[pessimistic-txn]` category. Besides `enable` and `default`, you can also configure the following parameters:
+
+- `ttl`
+
+    ```
+    ttl = "30s"
+    ```
+
+    `ttl` is the timeout time for the lock of pessimistic transactions. Its default value is "30s" (30 seconds). You must set it to a value between 15~60 seconds, and a lower or higher value can result in an error.
+
+    A transaction fails when its execution time exceeds `ttl`. If the value of `ttl` is set too high, the remaining pessimistic lock might block the write transaction for a long time when `tidb-server` is down. If set too low, the transaction might be rolled back by other transactions before it can finish execution.
+
+- `max-retry-count`
+
+    ```
+    max-retry-count = 256
+    ```
+
+    A pessimistic transaction can automatically retry a single statement. You can specify the maximum retrying times by setting the parameter `max-retry-count` to avoid retrying a statement endlessly in some extreme cases. Normally, you do not need to modify this configuration.
```

Comment:
```suggestion
A pessimistic transaction can automatically retry a single statement. You can specify the maximum retrying times by setting the `max-retry-count` parameter to avoid retrying a statement endlessly in some extreme cases. Normally, you do not need to modify this configuration.
```